.github/workflows/actions_docs.yml

```diff
@@ -0,0 +1,39 @@
+name: sphinx
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    # Standard drop-in approach that should work for most people.
+    - name: Sphinx Build
+        uses: ammaraskar/sphinx-action@0.4
+      with:
+        docs-folder: "docs/"
+
+    # Publish built docs to gh-pages branch.
+    # ===============================
+    - name: Commit documentation changes
+      run: |
+        git clone git@github.com:mehta-lab/microDL.git --branch gh-pages --single-branch gh-pages
+        cp -r docs/_build/html/* gh-pages/
+        cd gh-pages
+        touch .nojekyll
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git add .
+        git commit -m "Update documentation" -a || true
+        # The above command will fail if no changes were present, so we ignore
+        # that.
+    - name: Push changes
```

Comment:
Have you tested if this workflow will work?

Reply:
@ziw-liu It was something I didn't test because I didn't see a need to figure out how to checkout this branch after cloning the repo to github pages, since in the future this action will only happen when pushing to main and we only want to clone the main branch.
So I removed the actions_docs from this PR. I will submit a separate PR with this workflow once this PR is merged to main. Thank you for reviewing again.